### ISSUE TITLE
Agentless investigation

### ISSUE THREAD

**Issue by antoniomanuelfr** [wazuh/wazuh#14406]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.5 | Agentless | Manager/Agent | Packages/Sources | OS version |

Hi team!

This issue aims to collect a series of problems/issues in the agentless module.
Searching in the Github repository for the key `agentless` the following issues appear:

## Issues related to the handling of agentless "agents"
- https://github.com/wazuh/wazuh/issues/3302: Agentless entries are not removed. This issue has an associate PR adding some code to remove the dangling files: https://github.com/wazuh/wazuh/issues/3302. 
- https://github.com/wazuh/wazuh/issues/9165: agentless credentials stored in a file with weak permissions. There is a working issue where the `client.keys` file contents will be migrated to a database (https://github.com/wazuh/wazuh/issues/12075). This could be a good solution to store the ssh keys as long as only authorized users can query the DB. 

## `ssh_integrity_check` related issues
- https://github.com/wazuh/wazuh/issues/4287: ssh_integrity_check is not working in the latest versions of BSD. Also, this issue could be related to https://github.com/wazuh/wazuh/issues/8791. We will have to review the BSD script in charge of taking the information and make any necessary changes (some of these needed changes have been proposed in the issue comments).
- https://github.com/wazuh/wazuh/issues/8791 -> several agentless overrides each other states. As proposed in this issue, we should change the information flow in order to store the agentless information in different databases. A proposed solution could be the following:
![image](https://user-images.githubusercontent.com/22987701/180394438-9c720a3a-c4d0-40ce-b248-1b8dbf9a35d3.png)

## Crashes 
- https://github.com/wazuh/wazuh/issues/5649 -> `use_su` option leads to an invalid pointer, causing `agentlessd` to crash.

## Misc
- https://github.com/wazuh/wazuh/issues/2548 -> This issue can be closed as a duplicate issue.

Besides all these known issues, we have to keep in mind that agentless is using the [Standard OSSEC Message format](https://documentation.wazuh.com/current/development/message-format.html). Currently, the team is working on the new engine module, and probably, the message format should be also updated to match the new features of the engine module.

**Comment by antoniomanuelfr** [comment]
As part of this redesign, the interface could also be updated. 
As stated [here](https://documentation.wazuh.com/current/user-manual/capabilities/agentless-monitoring/agentless-faq.html#if-i-add-an-agentless-device-will-it-show-as-an-agent), the agentless agents do not appear as individual agents and in order to filter the information, the user needs to search in the `location` field the specific agent. 

A new menu could be added to the interface (or even in the agents menu) showing the agentless agents and their specific information. Having this information in a specific DB per agent will definitely help. Also, some API functions should be added in order to be able to create queries and retrieve information from Wazuh DB.

![image](https://user-images.githubusercontent.com/22987701/180416820-e3020c73-0e60-42f5-886a-b60a0c643cc1.png)

**Comment by hitman28594** [comment]
Hi,

Will this new proposal allow for tracking of remote syslog sources too (particularly those without ssh access). 

It would be nice to be able to keep track of these logsources. Often times network devices can be managed by third parties who don’t allow ssh access to the devices and the network devices only support syslog for logging. 

In these cases, it is VERY difficult to keep track and manage these “agentless and ssh-less devices” effectively. 

Thank you for your support :)

**Comment by antoniomanuelfr** [comment]
Hello @hitman28594 

The main approach that the agentless module uses for retrieving the information is by using ssh. I think that trying to use agentless on devices that are not running ssh or restricting the ssh connection is adding a use case that is going to add a lot of complexity to the module.

I think that in these cases, is better to use tools like rsyslog (https://www.rsyslog.com/) to forward the desired logs to the manager, and use logcollector to monitor the already forwarded logs.

**Comment by hitman28594** [comment]
@antoniomanuelfr 

Fair enough, although I don’t believe that this feature should limit itself to SSH being a requirement though. 

For example:
- we could run port scans against the agentless device to provide syscollector network information of the host.  


The vulnerability detector like feature for agentless devices has also been requested here:
https://github.com/wazuh/wazuh/issues/7076

**Comment by hitman28594** [comment]
And here: 
https://github.com/wazuh/wazuh/issues/13539

**Comment by hanschl** [comment]
Hello, 

is there already an ETA when to expect the fix? We are cuurently running version 4.9.0 and still have this issue as you can see in the screenshot.

![image](https://github.com/user-attachments/assets/9da89c69-d112-44c1-9244-837cecc97447)

Best Regards
Johannes Schlueter

**Comment by hanschl** [comment]
Hello,

could you please give an update to this issue?

Best Regards
Johannes Schlueter
